### ISSUE TITLE
Request: More detailed documentation about the configuration file

### ISSUE THREAD

**Issue by grammy-jiang** [codrsquad/portable-python#38]
This repository has a very high code quality and very decent output.

I notice this portable-python package is quite configurable, but I can't find any documentation about it, and have to read the source code and test cases. It is not very efficient and a little painful.

Is that possible to add more documentation about portable-python.yaml?

**Comment by solvingj** [comment]
Same request. 

Indeed, because of the quality of everything else, I figured the versions of deps had to be overridable in the config but no examples or samples indicate what the structure in the yaml should be. 

For posterity, it's just `<module_name>-version: <desired_version>` , for example:
````
openssl-version: 1.1.1w
````

But given this would be so simple to document, it was just outside the threshold of reasonability for non-python experts to have to piece this one together.

It made me wonder what other interesting config options might there be which people would use a lot of there was a sample config with all the possible keys/values stubbed out.

**Comment by zsimic** [comment]
You are right, I never got around to writing docs for this project. It is "wrapped up", but for use if you know how it works, the doc is not there :(

I need to carve out some time to come back to this, document it, and make the configuration a bit better as well. Especially to allow for things like this: https://github.com/codrsquad/portable-python/issues/7 (ie: allow source tarballs to come from an internal server, with sha checksums)
